crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -100,55 +244,341 @@ fn site_packages_dir_from_sys_prefix(
+            minor_version,
+            system,
+        )?];
+
+        if *include_system_site_packages {
+            let system_sys_prefix =
+                SysPrefixPath::from_executable_home_path(base_executable_home_path, system);
+
+            // If we fail to resolve the `sys.prefix` path from the base executable home path,
+            // or if we fail to resolve the `site-packages` from the `sys.prefix` path,
+            // we should probably print a warning but *not* abort type checking
+            if let Some(sys_prefix_path) = system_sys_prefix {
+                if let Ok(site_packages_dir) =
+                    site_packages_dir_from_sys_prefix(&sys_prefix_path, minor_version, system)
+                {
+                    site_packages_dirs.push(site_packages_dir);
+                }
+            }
+        }
+
+        tracing::debug!("Resolved site-packages directories for this virtual environment are: {site_packages_dirs:?}");
+        Ok(site_packages_dirs)
+    }
 }
 
 #[cfg(test)]
 mod tests {
-    use ruff_db::system::{OsSystem, System, SystemPath};
+    use ruff_db::system::{OsSystem, TestSystem};
 
-    use crate::site_packages::site_packages_dirs_of_venv;
+    use super::*;
 
-    #[test]
-    // Windows venvs have different layouts, and we only have a Unix venv committed for now.
-    // This test is skipped on Windows until we commit a Windows venv.
-    #[cfg_attr(target_os = "windows", ignore = "Windows has a different venv layout")]
-    fn can_find_site_packages_dir_in_committed_venv() {
-        let path_to_venv = SystemPath::new("resources/test/unix-uv-venv");
+    fn check_venv_basics_at_path(venv_name: impl AsRef<SystemPath>) {
+        let path_to_venv = SystemPath::new("resources/test").join(venv_name);
         let system = OsSystem::default();
 
         // if this doesn't hold true, the premise of the test is incorrect.
-        assert!(system.is_directory(path_to_venv));
+        assert!(system.is_directory(&path_to_venv));
+
+        let virtual_environment = VirtualEnvironment::new(path_to_venv.clone(), &system).unwrap();
+        assert_eq!(
+            virtual_environment.venv_path,
+            SysPrefixPath(system.canonicalize_path(&path_to_venv).unwrap())
+        );
+        assert!(!virtual_environment.include_system_site_packages);
+        assert!(virtual_environment
+            .minor_version
+            .is_some_and(|minor_ver| minor_ver == 12));
 
-        let site_packages_dirs = site_packages_dirs_of_venv(path_to_venv, &system).unwrap();
+        assert!(!virtual_environment
+            .base_executable_home_path
+            .as_str()
+            .is_empty());
+
+        let site_packages_dirs = virtual_environment.site_packages_dirs(&system).unwrap();
         assert_eq!(site_packages_dirs.len(), 1);
     }
+
+    // Windows venvs have different layouts, and we only have Unix venvs committed for now.
+    // These tests are skipped on Windows until we commit Windows venvs.
```

Comment:
I think it would be good to also have a windows venv. Considering that this is the platform that **we** use the least. Meaning, we won't get any coverage by using red knot ourselves.